mlflow/models/evaluation/default_evaluator.py

```diff
@@ -614,7 +662,53 @@ def plot_pr_curve():
 
         self._log_pandas_df_artifact(per_class_metrics_collection_df, "per_class_metrics")
 
-    def _evaluate_custom_metrics(self):
+    def _log_custom_metric_artifact(
+        self, artifact_name, raw_artifact, custom_metric_index, custom_metric_name
+    ):
+        if isinstance(raw_artifact, EvaluationArtifact):
+            return raw_artifact
+
+        exception_header = _get_custom_metric_exception_header(
+            custom_metric_index, custom_metric_name
+        )
+
+        inferred_from_path, inferred_type, inferred_ext = _infer_artifact_type_and_ext(
+            artifact_name, raw_artifact, exception_header
+        )
+        artifact_file_name = _gen_log_key(artifact_name, self.dataset_name) + inferred_ext
+        artifact_file_local_path = self.temp_dir.path(artifact_file_name)
+
+        if inferred_from_path:
+            shutil.copyfile(raw_artifact, artifact_file_local_path)
+        elif inferred_type is CsvEvaluationArtifact:
+            raw_artifact.to_csv(artifact_file_local_path, index=False)
+        elif inferred_type is NumpyEvaluationArtifact:
+            np.save(artifact_file_local_path, raw_artifact, allow_pickle=False)
+        elif inferred_type is ImageEvaluationArtifact:
+            raw_artifact.savefig(artifact_file_local_path)
+        else:
+            # try serializing as JSON.
+            try:
+                json.dump(raw_artifact, open(artifact_file_local_path, "w"))
+            except TypeError:
+                raise MlflowException(
+                    f"{exception_header} produced an unsupported artifact '{artifact_name}' with "
+                    f"type '{type(raw_artifact)}'. Supported object types for artifacts are:"
+                    f"- A string uri representing the file path to the artifact. MLflow"
+                    f"  will infer the type of the artifact based on the file extension."
+                    f"- EvaluationArtifact type objects. i.e. ImageEvaluationArtifact."
+                    f"- Pandas DataFrame. This will be resolved as a CSV artifact."
+                    f"- Numpy array. This will be saved as a .npy artifact."
+                    f"- Matplotlib Figure. This will be saved as an image artifact."
+                    f"- Any JSON serializable object. This will be saved as a JSON file."
+                )
+
+        mlflow.log_artifact(artifact_file_local_path)
+        artifact = inferred_type(uri=mlflow.get_artifact_uri(artifact_file_name))
+        artifact._load(artifact_file_local_path)
+        return artifact
+
+    def _log_custom_metrics(self):
```

Comment:
custom metrics are not logged in this function (it's done in`self. _log_metrics`). What happened inside is "evaluate custom functions" and "log custom artifacts". I suggest to separate this into 2 functions to make it really clear what's been done in each. Or at least name it accurately if separating the logic is complicated.